ghproxy/ghcache/partitioner.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ghcache
+
+import (
+	"net/http"
+	"sync"
+)
+
+type roundTripperCreator func(partitionKey string) http.RoundTripper
+
+// partitioningRoundTripper is a http.RoundTripper
+var _ http.RoundTripper = &partitioningRoundTripper{}
+
+func newPartitioningRoundTripper(rtc roundTripperCreator) *partitioningRoundTripper {
+	return &partitioningRoundTripper{
+		roundTripperCreator: rtc,
+		lock:                &sync.Mutex{},
+		roundTrippers:       map[string]http.RoundTripper{},
+	}
+}
+
+type partitioningRoundTripper struct {
+	roundTripperCreator roundTripperCreator
+	lock                *sync.Mutex
+	roundTrippers       map[string]http.RoundTripper
+}
+
+func getCachePartition(r *http.Request) string {
+	return r.Header.Get("Authorization")
+}
+
+func (prt *partitioningRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	cachePartition := getCachePartition(r)
+
+	prt.lock.Lock()
+	roundTripper, found := prt.roundTrippers[cachePartition]
+	if !found {
+		cachePartitionsCounter.WithLabelValues(cachePartition).Add(1)
```

Comment:
Doesn't this put the paintext auth header into the prom label?

Reply:
Oups